src/api/kotlin/xyz/wagyourtail/unimined/util/FlexVerUtil.kt

```diff
@@ -0,0 +1,168 @@
+package xyz.wagyourtail.unimined.util
+
+import org.jetbrains.annotations.VisibleForTesting
+import java.util.*
+import kotlin.math.max
+import kotlin.math.min
+
+/**
+ * Parse the given strings as freeform version strings, and compare them according to FlexVer.
+ * @param b the second version string
+ * @return `0` if the two versions are equal, a negative number if `a < b`, or a positive number if `a > b`
+ */
+fun String.compareFlexVer(b: String): Int {
+    return FlexVerComparator.compare(this, b)
+}
+
+/**
+ * Implements FlexVer, a SemVer-compatible intuitive comparator for free-form versioning strings as
+ * seen in the wild. It's designed to sort versions like people do, rather than attempting to force
+ * conformance to a rigid and limited standard. As such, it imposes no restrictions. Comparing two
+ * versions with differing formats will likely produce nonsensical results (garbage in, garbage out),
+ * but best effort is made to correct for basic structural changes, and versions of differing length
+ * will be parsed in a logical fashion.
+ *
+ * @author Unascribed
+ */
+object FlexVerComparator {
+    /**
+     * Parse the given strings as freeform version strings, and compare them according to FlexVer.
+     *
+     * @param a the first version string
+     * @param b the second version string
+     * @return `0` if the two versions are equal, a negative number if `a < b`, or a positive number if `a > b`
+     */
+    fun compare(a: String, b: String): Int {
+        val ad = decompose(a)
+        val bd = decompose(b)
+        for (i in 0 until max(ad.size.toDouble(), bd.size.toDouble()).toInt()) {
```

Comment:
why are there toDouble's here, that doesn't seem right

also, maybe consider PR'ing this upwards to unascribed as kotlin-multiplatform

Reply:
The idea behind adding it directly is to not require shadowing the library (it would be the first) or adding another maven repository the the list (which would propagate to every dependent project).